src/main/java/com/teragrep/pth_06/planner/bloomfilter/BloomFilterFromRecord.java

```diff
@@ -0,0 +1,147 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner.bloomfilter;
+
+import org.apache.spark.util.sketch.BloomFilter;
+import org.jooq.Record;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import org.jooq.types.ULong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.teragrep.pth_06.jooq.generated.bloomdb.Bloomdb.BLOOMDB;
+
+/**
+ * Extracts filter type from record, creates a bloom filter and returns the filters byte array
+ */
+public final class BloomFilterFromRecord {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(BloomFilterFromRecord.class);
+    private final Long expected;
+    private final Double fpp;
+    private final String pattern;
+    private final String searchTerm;
+
+    private BloomFilter create() {
+        if (expected == null || fpp == null) {
+            LOGGER
+                    .error(
+                            "Null field while creating bloom filter expected <{}>, fpp <{}>, pattern <{}>, search term <{}>",
+                            expected, fpp, pattern, searchTerm
+                    );
+            throw new RuntimeException("Object field was null");
```

Comment:
this exception message could be a bit clearer

Reply:
Clarified the exception messages, added tests for excetpions, removed use of  `.longValue()` method in constructor which would lead to NPE.